### PR TITLE
Create hooks folder if it doesn't exist

### LINKED ISSUE #13: hook install fails
npm install jira-precommit-hook is failing to install
image

### PR DESCRIPTION
**Pull request by kthompson55**
Closes #13 by making sure a .git/hooks folder exists in the git repository.

### CODE REVIEW

**Comment by mtscout6** [comment]
Is there any way to add tests for this?

**Comment by kthompson55** [comment]
@mtscout6 I'll find a way.

File: test/fs-utils-test.js [hunk 1]
```diff
@@ -16,6 +16,22 @@ describe('FS-Utils Tests', () => {
...
+        if(exists) {
+          fsp.rmdir(path.join('test', 'tmp', '.git', 'hooks'));
+        }
+      })
+    );
+
+    it('Confirm hooks folder exists', done => {
+      verifyHooksFolder(path.join('test', 'tmp', '.git', 'hooks'));
+      fs.existsSync(path.join('test', 'tmp', '.git', 'hooks')).should.equal(true);
+      done();
```
Comment (mtscout6) [inline test/fs-utils-test.js]: Why is this an async test? I don't see anything async about it.

File: src/fs-utils.js [hunk 2]
```diff
@@ -23,6 +23,13 @@ export function findParentFolder(startDir, parentDirName) {
   return currentDir;
 }
 
+export function verifyHooksFolder(desiredHooksPath) {
+  if(!fs.existsSync(desiredHooksPath)) {
+    console.log('Creating hooks directory in .git');
+    fs.mkdir(desiredHooksPath);
```
Comment (mtscout6) [inline src/fs-utils.js]: You are using a mix of sync and async methods here, `fs.mkdir` is async. Please be consistent, and return a promise if you go with the async approach. (I think synchronous is fine here). See: https://nodejs.org/api/fs.html#fs_fs_mkdir_path_mode_callback

**Comment by aabenoja** [comment]
1. You can test `verifyHooksFolder` directly with stubs and spies.
2. You created `rmdirRecursively`, but it doesn't look like it's being used.

**Comment by kthompson55** [comment]
@aabenoja `rmdirRecursively` is used inside of `fs-utils-test.js`

File: src/fs-utils.js [hunk 3]
```diff
@@ -23,6 +23,27 @@ export function findParentFolder(startDir, parentDirName) {
...
 }
 
+export function verifyHooksFolder(desiredHooksPath) {
+  if(!fs.existsSync(desiredHooksPath)) {
+    console.log('Creating hooks directory in .git');
+    fs.mkdirSync(desiredHooksPath);
+  }
+}
+
+export function rmdirRecursively(directory) {
```
Comment (mtscout6) [inline src/fs-utils.js]: You could use `rimraf` for this.

Comment (mtscout6) [inline src/fs-utils.js]: https://www.npmjs.com/package/rimraf

Comment (aabenoja) [inline src/fs-utils.js]: :+1: `rimraf's great for this.

Comment (kthompson55) [inline src/fs-utils.js]: I'll give it a try.

Comment (kthompson55) [inline src/fs-utils.js]: `rimraf` worked marvelously. The test that used it passed.

**Comment by aabenoja** [comment]
LGTM

**Comment by mtscout6** [comment]
LGTM

**Pull request merged**